editor/src/messages/tool/tool_messages/pen_tool.rs

```diff
@@ -399,10 +401,11 @@ impl Fsm for PenToolFsmState {
 					tool_data.should_mirror = true;
 					process().unwrap_or(PenToolFsmState::PlacingAnchor)
 				}
-				(PenToolFsmState::DraggingHandle, PenToolMessage::PointerMove { snap_angle, break_handle }) => {
+				(PenToolFsmState::DraggingHandle, PenToolMessage::PointerMove { snap_angle, break_handle, lock_angle }) => {
 					let mut process = || {
 						// Get subpath
-						let layer_path = tool_data.path.as_ref()?;
+						let data = tool_data.clone();
+						let layer_path = data.path.as_ref()?;
```

Comment:
Is there a better way to deal with the exclusive tool_data ref being borrowed immutably here?

Reply:
Couldn't you just move the angle computation above the layer_path declaration? From the looks of it you only need a mutable ref for the computation and can safely demote it to a shared ref afterwards.

Reply:
I'm not sure that will be possible. The calculation needs `last_anchor.position`, which is worked out something like this: `tool_data -> layer_path -> subpath -> manipulator_groups -> last_anchor`.

Reply:
Instead of passing `tool_data` to the `compute_snapped_angle`, just pass `&mut tool_data.angle` (rust allows you to borrow different fields of a struct at the same time).